packages/devtools_app/lib/src/timeline/flutter/timeline_controller.dart

```diff
@@ -275,14 +276,23 @@ class TimelineController implements DisposableController {
     _loadOfflineDataController.add(offlineTimelineData);
   }
 
-  int _threadIdForEvent(
-    String targetEventName,
+  int _threadIdForEvents(
+    List<String> targetEventNames,
     List<TraceEventWrapper> traceEvents,
   ) {
     const invalidThreadId = -1;
     return traceEvents
-            .firstWhere((trace) => trace.event.name == targetEventName,
-                orElse: () => null)
+            .firstWhere(
+              (trace) {
+                for (final target in targetEventNames) {
```

Comment:
And here, return `targetEventNames.contains(target.event.name)` (which will be efficient, given the change to a Set).

Reply:
done